### PR TITLE
feature/api-search

### PR DESCRIPTION
**Pull request by jrouzierinverse**
Description
===========
Search API

Impacts
=======
pfunified_api

Delete branch after merge
=========================
NO

### CODE REVIEW

**Comment by julsemaan** [comment]
If its ready for review, could you mark it as so with the tags.

Just so I don't comment on it for stuff you're already fixing

**Comment by extrafu** [comment]
@julsemaan Looks good? The idea is to merge this quickly so we can start using it from the GUI.

We can fix any issue that arise directly in devel.

File: t/unittest/UnifiedApi/Controller/Locationlogs.t [hunk 1]
```diff
@@ -73,16 +73,214 @@ $t->get_ok('/api/v1/locationlogs' => json => { })
...
+    '/api/v1/locationlogs/search' => json => { limit => 267 }
+  )
+  ->status_is(200)
+  ->json_hasnt('/nextCursor')
+  ->json_is('/prevCursor', 0)
+;
+$items = $t->tx->res->json->{items};
+
+is(@$items, 256, "Returns 256 items");
+
```
Comment (julsemaan) [inline t/unittest/UnifiedApi/Controller/Locationlogs.t:283]: Maybe I'm blind but we're missing a test for using the cursor parameter for pagination

**Review by julsemaan — COMMENTED** [review]
Just make sure that pagination is well tested in search and we're good for merge

File: t/unittest/UnifiedApi/Controller/Locationlogs.t (hunk 1 above)
Comment (jrouzierinverse) [reply t/unittest/UnifiedApi/Controller/Locationlogs.t:283]: https://github.com/inverse-inc/packetfence/pull/2992/files/4b0d2a630f5e269803c225d5dc740ca63d359413#diff-acbd078bd53801f2f2fc209e5c36a09fR230

**Pull request merged**